spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/GlobalKTableBoundElementFactory.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kafka.streams;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.apache.kafka.streams.kstream.GlobalKTable;
+
+import org.springframework.aop.framework.ProxyFactory;
+import org.springframework.cloud.stream.binder.ConsumerProperties;
+import org.springframework.cloud.stream.binding.AbstractBindingTargetFactory;
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+import org.springframework.util.Assert;
+
+/**
+ * {@link org.springframework.cloud.stream.binding.BindingTargetFactory} for {@link GlobalKTable}
+ *
+ * Input bindings are only created as output bindings on GlobalKTable are not allowed.
+ *
+ * @author Soby Chacko
+ */
+public class GlobalKTableBoundElementFactory extends AbstractBindingTargetFactory<GlobalKTable> {
+
+	private final BindingServiceProperties bindingServiceProperties;
+
+	GlobalKTableBoundElementFactory(BindingServiceProperties bindingServiceProperties) {
+		super(GlobalKTable.class);
+		this.bindingServiceProperties = bindingServiceProperties;
+	}
+
+	@Override
+	public GlobalKTable createInput(String name) {
+		ConsumerProperties consumerProperties = this.bindingServiceProperties.getConsumerProperties(name);
+		//Always set multiplex to true in the kafka streams binder
```

Comment:
Is there some explanation "Why?" ? And from here: why isn't it by default in the Kafka Streams per se since it is so important for us?

Thanks

Reply:
Since it is allowed to consume from multiple topics in a single binding, multiplex is enabled by default in the Kafka streams binder. In the regular Kafka binder, the user has to explicitly enable it. See this [javadoc](https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/ConsumerProperties.java#L144). As you suggested, It definitely makes sense to enable it by default across the various binding types in Kafka streams binder (`KStream`, `KTable` and `GlobalKTable`) instead of doing it individually.

Reply:
That's not what I'm asking about, but let it be.
I believe if this comment would be much informative, I'd not ask this question.